### ISSUE TITLE
Add more possibilities for modules

### ISSUE THREAD

**Issue by TracerDS** [multitheftauto/mtasa-blue#2987]
### Is your feature request related to a problem? Please describe.

Calling lua functions from C just to call C function is ridiculous.
e.g. in order to add an event you have to push some values on lua stack and call `addEvent` function which calls mta function in C++.
Adding references to mta classes (CEvents, CLuaMain, CPlayer, CElement, etc.) in `ILuaModuleManager` would make the whole process easier (and probably faster since you dont need to switch context)

### Describe the solution you'd like

Add `CGame`, `CEvents`, `CPedManager`, etc. references to the `ILuaModuleManager` class.

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not about a cheat or security vulnerability.

**Comment by CrosRoad95** [comment]
> Calling lua functions from C just to call C function is ridiculous.

suggested it in lua-modules channel but i will do the same here.
What if we allow to load modules that are written in lua with highly privilaged set of functions that let you for example modify any resource before starting?

**Comment by TracerDS** [comment]
> > Calling lua functions from C just to call C function is ridiculous.
> 
> suggested it in lua-modules channel but i will do the same here. What if we allow to load modules that are written in lua with highly privilaged set of functions that let you for example modify any resource before starting?

yeah I wanted to create a separate issue for that to make things easier.

lua modules might actually complicate everything since you'd only have full access to the server and no new features could be added via them
